Automattic/wp-calypso
# Fix the flickering of upsell banners in /hosting-config

Fixes https://github.com/Automattic/dotcom-forge/issues/9531

## Proposed Changes

* Set the loading state to null to indicate that the data has never been fetched before
* Only show the upsell banner after we are sure we've loaded from the API

#### Before

https://github.com/user-attachments/assets/53c33a69-dbfa-45ce-853b-4cfa1b1dea4a



#### After

https://github.com/user-attachments/assets/156617a6-08aa-4966-9560-5b394ae1fd03



## Why are these changes being made?


* Prevent flickering

## Testing Instructions



* Go to `/hosting-config/:site`, note that you may see different navigation tabs on wpcalypso and dev environment, you can turn it off by adding `?flags=-untangling/hosting-menu`, this is part of pbxlJb-6ye-p2
* Check if the banner flickers
* Test with staging sites, e-commerce trial sites, and atomic sites to make sure we've fixed it

> [!NOTE]  
> We will not aim to improve is the SSH/SFTP card loading state due to the reasons [mentioned here](https://github.com/Automattic/dotcom-forge/issues/9531#issuecomment-2462202759). And with the PR we've solved the original concern.


## Pre-merge Checklist
